packages/ui/src/proposals/modals/AddNewProposal/components/SpecificParameters/FundingRequest.tsx

```diff
@@ -1,13 +1,51 @@
-import React from 'react'
+import React, { useCallback, useEffect, useState } from 'react'
+import { useFormContext } from 'react-hook-form'
+import styled from 'styled-components'
 
 import { SelectAccount } from '@/accounts/components/SelectAccount'
 import { CurrencyName } from '@/app/constants/currency'
-import { InputComponent, TokenInput } from '@/common/components/forms'
+import { ButtonPrimary } from '@/common/components/buttons'
+import {
+  InlineToggleWrap,
+  InputComponent,
+  Label,
+  ToggleCheckbox,
+  TokenInput,
+  InputTextarea,
+} from '@/common/components/forms'
+import { Arrow } from '@/common/components/icons'
 import { Row } from '@/common/components/Modal'
 import { RowGapBlock } from '@/common/components/page/PageContent'
-import { TextMedium } from '@/common/components/typography'
+import { Tooltip, TooltipDefault } from '@/common/components/Tooltip'
+import { TextMedium, TextSmall, TextInlineSmall } from '@/common/components/typography'
+
+import { PreviewAndValidateModal } from './modals/PreviewAndValidate'
+import { ErrorPrompt, Prompt } from './Prompt'
 
 export const FundingRequest = () => {
+  const { watch, setValue, getValues, getFieldState } = useFormContext()
+  const [isPreviewModalShown, setIsPreviewModalShown] = useState(false)
+  const [payMultiple] = watch(['fundingRequest.payMultiple'])
+  const [hasPreviewedInput] = watch(['fundingRequest.hasPreviewedInput'], { 'fundingRequest.hasPreviewedInput': true })
+  const csvInput = watch('fundingRequest.csvInput')
+  useEffect(() => {
+    if (getFieldState('fundingRequest.accountsAndAmounts')) {
+      setValue('fundingRequest.accountsAndAmounts', undefined, { shouldValidate: true })
+      setValue('fundingRequest.hasPreviewedInput', false, { shouldValidate: true })
+    }
+  }, [csvInput])
+
+  const shouldDisablePreviewButton = () => {
+    const input = getFieldState('fundingRequest.csvInput')
+    return input.error !== undefined || !input.isDirty
+  }
```

Comment:
Since this function is called on every render, it would be simpler to just have a variable:
 ```suggestion
  const canPreviewInput = csvInput.isDirty && !csvInput.error
```